src/Microsoft.OData.Core/UriParser/Parsers/FunctionCallParser.cs

```diff
@@ -77,9 +100,22 @@ public QueryToken ParseIdentifierAsFunction(QueryToken parent)
                 this.Lexer.NextToken();
             }
 
-            FunctionParameterToken[] arguments = this.ParseArgumentListOrEntityKeyList();
+            try
+            {
+                FunctionParameterToken[] arguments = this.ParseArgumentListOrEntityKeyList();
+                result = new FunctionCallToken(functionName, arguments, parent);
+            }
+            catch (ODataException e)
+            {
+                if (!restoreStateIfFail)
+                {
+                    throw e;
+                }
+
+                lexer.RestorePosition(position);
+            }

```

Comment:
Is there a way to execute the "try" without using exceptions as a form of communication?